### ISSUE TITLE
Mac OS X Support Roadmap

### ISSUE THREAD

**Issue by joshgarde** [NodeOS/NodeOS#322]
So a few months ago I got started on OS X build support for NodeOS, but I never really finished the effort. I've finally found the time to come back and revisit it.

Here's a summary of what installs/builds and what doesn't

- [ ] NodeOS
- [ ] nodeos-barebones
- [x] nodeos-bootfs [PARTIAL]
- [x] nodeos-cross-toolchain [PARTIAL]
- [ ] nodeos-initramfs
- [ ] nodeos-userfs

Mac OS X is an odd beast to port over to, but progress will slowly be made. It's nice to be back here.

**Comment by piranna** [comment]
By the way, both `nodeos-cross-toolchain` and `nodeos-usersfs` are mostly autonomous and frozen regarding the new architecture, so maybe you would be interested in take a look at them for the porting :-) I hope to have the others ready for this weekend :-)

**Comment by joshgarde** [comment]
Any updates on the refactor?

**Comment by piranna** [comment]
> Any updates on the refactor?

It's almost done...

Prebuild images are being generated, but it was flaky because the partitions where hardcoded. I've been all the weekend fighting with `libblkid` because its code sucks, but finally I was able to get the dev path from the partition UUID, problem is that only when used as dynamic library and not compiled statically because the library is difficult to configure to be used with `gyp`, so it will be needed to bea added. Ugly, but if it works, it would be enough by now...

After that, the tests are not passing due to `nsh` and the Node.js bug regarding stdio redirection. I'm thinking about using a less strict test so they can pass and later focus on that problem, doing so probably I would be able to release it today or tomorrow, what do you think?

**Comment by joshgarde** [comment]
Take your time. I'm going to get started on porting over node-userfs in the meanwhile.

Just a quick question: Why are a lot of NodeOS packages using version numbers that are breaking builds? For example: https://github.com/NodeOS/nodejs

**Comment by joshgarde** [comment]
`nodeos-cross-toolchain` prebuild uploaded

**Comment by joshgarde** [comment]
Also, can I get permission to upload prebuilds to repos?

**Comment by joshgarde** [comment]
No, I don't have access to the other repos. As for automation, I might just experiment with running OSX on traditional Linux VPSs for cheaper overall costs.

**Comment by piranna** [comment]
> No, I don't have access to the other repos

Ok, I could be able to uploade them myself, or I could be able to give you access on request.


> As for automation, I might just experiment with running OSX on traditional Linux VPSs for cheaper overall costs

I would stick with TravisCI, at least we can be able to do some checkings and do a basic build although other targets would be needed to be compiled by the users, the process is automated. If we need other targets too, we can be able to publish them by hand.

**Comment by joshgarde** [comment]
Can I get TravisCI enabled on qemu? Prebuilds should work there.

**Comment by piranna** [comment]
> Can I get TravisCI enabled on qemu? Prebuilds should work there.

Well... [it's already enabled](https://travis-ci.org/NodeOS/qemu), only that they are not being published. It should be easy, though...
